### ISSUE TITLE
Add model/material/hierarchy and information

### ISSUE THREAD

**Issue by mwestphal** [f3d-app/f3d#18]
more info here: https://gitlab.kitware.com/f3d/f3d/-/issues/174

**Comment by AmazingMrX** [comment]
I'd like to see this model hierchy added just so users can toggle visibility of the model components. Right now, F3D renders everything by default. However, complex models may be designed to only be viewed in an arbitrary configuration of components that's only a subset of all components present in the file. 

Example: An FBX file containing a character model with multiple attached alternative outfits.

It may be possible that this could apply to materials and textures as well. Say, for instance: An object in a video game has multiple states represented by differing textures that are swapped programmatically.

It would be nice to accurately view these assets in F3D with some simple visibility toggles on a hierarchical list.

**Comment by AmazingMrX** [comment]
Yep! [Here](https://free3d.com/3d-model/low-poly-isometric-room-1-704614.html) is a freely available model of a living room in a variety of formats.
I've got the FBX version temporarily hosted [here](https://autode.sk/42acCYw) on Autodesk's online Viewer (with textures) so you can readily see the hierarchy chart. Click "Model Browser" in the upper left to see the list with everything that can be toggled in the file. I'm basically asking for the same functionality in F3D.

**Comment by mwestphal** [comment]
Ok I see that. We need to check what assimp (what we are using to read fbx) is providing us, then we need to see how to expose some kind of block selection mechanism. That may come at some point but this is likely in the long term feature, definitely after #653 .

Wdyt @Meakk ?

**Comment by mwestphal** [comment]
@AmazingMrX please create a dedicated issue for your idea :)

**Comment by Medyan-Naser** [comment]
Hi @mwestphal,

I can work on this issue, please assign it to me.
Also, is there any update on this issue, or do you have any suggestions for what to look for.

**Comment by mwestphal** [comment]
Hi @Medyan-Naser 

This one is a big one! however you should be able to do the first step using GLTF files as the GLTF importer does implement the scene API.

I can create a dedicated issue giving more details, does that sound good to you ?

**Comment by Medyan-Naser** [comment]
Yes sounds good. I am just planning on closing the parent issue.

**Comment by mwestphal** [comment]
@Meakk this can be closed right ?

**Comment by Meakk** [comment]
Yes but we still need to add apis and scene hierarchy for assimp/usd/quake importers

**Comment by mwestphal** [comment]
Created the follow up issues:
 - https://github.com/f3d-app/f3d/issues/2948
 - https://github.com/f3d-app/f3d/issues/2949
 - https://github.com/f3d-app/f3d/issues/2950
